activerecord/test/cases/associations/eager_test.rb

```diff
@@ -1340,6 +1339,10 @@ def test_deep_including_through_habtm
     # has-many :through
     david = Author.where(id: "1").eager_load(:readonly_comments).first!
     assert david.readonly_comments.first.readonly?
+
+    # belongs_to
+    post = Post.where(id: "1").eager_load(:author).first!
```

Comment:
It seems that `joins` doesn’t make the association `readonly`, while `eager_load` does. I’ve opened a bug #24093.